src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java

```diff
@@ -148,6 +200,10 @@ public void moveMonster(){
 		    } else if (logicHandler.movable(xPos, yPos)) {
 			logicHandler.move(xPos, yPos,xOrig,yOrig);
 			canvas.moveMonster(xPos, yPos,logicHandler.getObject(xPos,yPos));
+		    }else if ((!logicHandler.movable(xPos,yPos))){
+			logicHandler.move(xOrig, yOrig,xOrig,yOrig);
+			canvas.moveMonster(xOrig, yOrig,logicHandler.getObject(xOrig,yOrig));
```

Comment:
And what's the point of "moving" a monster to the same slot?

Reply:
This was for monster disappearing

Reply:
This seems like a bandaid over a more fundamental problem...  What's the actual problem, and how does this fix it?